### ISSUE TITLE
Support custom optimizer with loss + model parameters in TabNetRegressor

### ISSUE THREAD

**Issue by starlight345** [dreamquark-ai/tabnet#574]
## Feature request

**What is the expected behavior?**

I would like to use a custom training loop with `TabNetRegressor`, where I define my own optimizer (e.g., combining model and loss function parameters), custom loss function (e.g., uncertainty-weighted loss), and train the TabNet model along with other trainable components.

This requires accessing the internal TabNet model (e.g., via `model.network` or `model.tabnet`) to retrieve its parameters and set it to training mode (`train()`), but it seems those attributes are not publicly accessible in the current version.

**What is motivation or use case for adding/changing the behavior?**

In many use cases, especially for multi-target regression or multitask learning, we want to:
- Define trainable loss functions that require joint optimization with the model.
- Use fine-tuned optimizers and learning rate schedules.
- Integrate TabNet with other modules (e.g., transformers, uncertainty layers) in a custom training loop.

While `TabNetRegressor` supports `fit()` using built-in optimizers, it limits flexibility for such advanced use cases.

**How should this be implemented in your opinion?**

Make the internal TabNet model (e.g., `self.network` or `self.tabnet`) accessible via a public attribute such as `model.tabnet` or `model.network`, or provide an official method to get the core TabNet module and its parameters for external training.

This would allow users to:

```python
optimizer = torch.optim.Adam(
    list(model.tabnet.parameters()) + list(loss_fn.parameters())
)
```
and run custom training loops while still using the high-level wrappers for initialization or pretraining.

**Are you willing to work on this yourself?**
yes

**Comment by Optimox** [comment]
If the current customizable pipeline is too restrictive for your use case, you can easily retrieve only the tabnet architecture from here: https://github.com/dreamquark-ai/tabnet/blob/2c0c4ebd2bb1cb639ea94ab4b11823bc49265588/pytorch_tabnet/tab_network.py#L508

As simple as : `from tabnet.tab_network import TabNet`
